tests/authentication/test_authentication.py

```diff
@@ -55,26 +56,70 @@ def test_for_missing_email(self):
                 "password": "akrammukasa"}}
         response = self.client.post('/api/users/', data, format='json')
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
-        self.assertIn("This field may not be blank.", str(response.data))
+        self.assertIn('Please enter a valid email address',
+                      response.data["errors"]["Email"])
 
     def test_login(self):
         data = {"user": {
                 "username": "akram",
                 "email": "akram@gmail.com",
-                "password": "akrammukasa"}
+                "password": "aKrammukasa1234"}
                 }
         response = self.client.post('/api/users/', data, format='json')
 
-        data = {"user": {"email": "akram@gmail.com", "password":
-                "akrammukasa"}}
-        response = self.client.post('/api/users/login/', data, format='json')
+        login_data = {"user": {"email": "akram@gmail.com", "password":
+                               "aKrammukasa1234"}}
+        self.client.post('/api/users/verify/?token=' + response.data['token'])
+        response = self.client.post(
+            '/api/users/login/', login_data, format='json')
```

Comment:
Please create custom methods to handle registration, verification, and log in. instead you calling them through many requests in a single test.